homeassistant/components/enocean/cover.py

```diff
@@ -0,0 +1,267 @@
+"""Support for EnOcean roller shutters (EEP D2-05-00)."""
+from __future__ import annotations
+
+import asyncio
+import logging
+
+from enocean.protocol.packet import Packet  # pylint: disable=import-error
+from enocean.utils import combine_hex  # pylint: disable=import-error
+import voluptuous as vol  # pylint: disable=import-error
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DEVICE_CLASS_BLIND,
+    DEVICE_CLASSES_SCHEMA,
+    PLATFORM_SCHEMA,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.const import CONF_DEVICE_CLASS, CONF_ID, CONF_NAME
+from homeassistant.core import HomeAssistant
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import dispatcher_send
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+
+from .const import SIGNAL_SEND_MESSAGE
+from .device import EnOceanEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "EnOcean roller shutter"
+DEPENDENCIES = ["enocean"]
+
+CONF_SENDER_ID = "sender_id"
+WATCHDOG_TIMEOUT = "watchdog_timeout"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
+        vol.Required(CONF_SENDER_ID): vol.All(cv.ensure_list, [vol.Coerce(int)]),
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+        vol.Optional(WATCHDOG_TIMEOUT, default=5): int,
+    }
+)
+
+
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Cover platform for EnOcean."""
+    dev_id = config.get(CONF_ID)
+    sender_id = config.get(CONF_SENDER_ID)
+    dev_name = config.get(CONF_NAME)
+    device_class = config.get(CONF_DEVICE_CLASS)
+    if device_class is None:
+        device_class = DEVICE_CLASS_BLIND
+    watchdog_timeout = config.get(WATCHDOG_TIMEOUT)
+    add_entities(
+        [
+            EnOceanCover(
+                sender_id, dev_id, dev_name, device_class, hass, watchdog_timeout
+            )
+        ]
+    )
+
+
+class EnOceanCover(EnOceanEntity, CoverEntity):
+    """Representation of an EnOcean Cover (EEP D2-05-00)."""
+
+    def __init__(
+        self, sender_id, dev_id, dev_name, device_class, hass, watchdog_timeout
```

Comment:
Don't pass in `hass`. It will be set on the entity when the entity has been added to home assistant. This means you can access `hass` with `self.hass`, after the entity has been added to home assistant.

Reply:
Thanks - see commit [c85fc44](https://github.com/home-assistant/core/pull/75356/commits/c85fc443ed5a0370857bc1fd76fcba3c9f723799)